### PR TITLE
Added test case for Icon.bind method

### PR DESCRIPTION
**Pull request by Apoorv-DTU**
Added a test case for the bind method of the Icon widget interface. It increases the test coverage by 4 lines.
## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/core/tests/widgets/test_icon.py [hunk 1]
```diff
@@ -9,4 +10,10 @@ def setUp(self):
...
 
-        self.icon = toga.Icon(self.text, factory=self.factory)
\ No newline at end of file
+        self.test_path = "Example.bmp"
+        self.icon = toga.Icon(self.test_path)
+
+    def test_icon_bind(self):
+        self.assertEqual(self.icon._impl, None)
+        self.icon.bind(factory=toga_dummy.factory)
+        self.assertEqual(self.icon._impl.interface, self.icon)
```
Comment (freakboy3742) [inline src/core/tests/widgets/test_icon.py]: It would also be a good idea to check that the bound icon is actually the same one that was passed in.

**Comment by freakboy3742** [comment]
I'm not completely sure what you've done here, but it looks like you've imported a bunch of commits, but *not* as a merge commit. As a result, it's now very difficult to tell what is a new change, and what is a change that has already been merged into master. Is it possible you've merged in the wrong direction?

**Comment by Apoorv-DTU** [comment]
I'm extremely sorry. I had no idea that if I pushed something to my forked repo, it would show up here automatically. None of those commits implements the desired changes. I will submit the required commit after some time today.

**Comment by freakboy3742** [comment]
@Apoorv-DTU No problems at all! Git and Github have a bit of a learning curve, and the feature you've just accidentally discovered actually makes some other things about working with Github much easier (once you're doing them on purpose, rather than accidentally!) Let us know when you're ready for another review, and I'll take a look.

**Comment by Apoorv-DTU** [comment]
Thank you for understanding! I have just pushed the commit implementing the changes. Please take a look at it.

**Comment by freakboy3742** [comment]
Ok - so, the problem is that the commit history is now corrupted - all the commits from other developers that you've merged in are showing up as *new* contributions in this patch.

I'm not 100% sure how to correct what's happened here (it's difficult to tell without knowing what you did). I suspect you may have gotten tied in knots dealing with the `origin` repository (the representation of *your* repository on Github) and the `upstream` repository (the Pybee version of the same repository). 

The best option I can think of is to rebase your branch against pybee/toga's master branch. That will re-write the history of your branch so that your changes are at the end. To do this, get the name of the remote that represents the main Pybee repository - you can find this by using `git remote -v`:

    $ git remote -v
    origin	git@github.com:freakboy3742/toga (fetch)
    origin	git@github.com:freakboy3742/toga (push)
    upstream	git@github.com:pybee/toga (fetch)
    upstream	git@github.com:pybee/toga (push)

Based on this output, `origin` is my own version of the repository, and `upstream` is Pybee's copy. From there, you can run::

    git fetch upstream
    git rebase upstream/master

Then, run `git push --force` to update this pull request. This will rewrite the pull request to remove the extra content.

If you *don't* have an `upstream` repository (or you've *only* got an origin), then that possibly points to the source of your problem. Run:

    git remote add upstream git@github.com:pybee/toga
    git fetch upstream

and then try again. This will add a reference to the main pybee repository into your own; in future, you can then run:

    git fetch upstream
    git merge upstream/master

to pull all the most recent changes from upstream into your own repository (or branch).

Let me know if you have any problems.

**Comment by Apoorv-DTU** [comment]
I hope it can be merged now.

**Pull request merged**